docs(chart): Add documentation for range series types (#1536)
telerik/blazor-docs: components/chart/types/rangebar.md

---
title: Range Bar
page_title: Chart - Range Bar

> description: Overview of the Range Bar Chart for Blazor.

Reviewer: Try to write descriptions that are around 150 symbols and describe the content of the article.

> description: Overview of the Range Bar Chart for Blazor.

Author: ```suggestion
description: Overview of the Range Bar Chart for Blazor with a description of the common use cases and the different ways to data bind the chart. The article lists configuration options and provides Range Bar Chart examples.
```

slug: components/chart/types/rangebar
tags: telerik,blazor,chart,rangebar
published: True
position: 0
---

# Range Bar Chart

> The <a href="https://www.telerik.com/blazor-ui/range-bar-chart" target="_blank">Blazor Range Bar chart</a> displays data as horizontal bars whose position and length vary according to their values. You can use a Range Bar chart to show a comparison between several sets of data (for example, summaries of quantative or time data). Each series is automatically colored differently for easier reading. The Range Bar Chart is similar to the [Bar Chart]({%slug components/chart/types/bar%}). The latter can be regarded as a Range Bar Chart with a "low" ("from") value of zero.

Reviewer: ```suggestion
The <a href="https://www.telerik.com/blazor-ui/range-bar-chart" target="_blank">Blazor Range Bar Chart</a> displays data as horizontal bars whose position and length vary according to their values. You can use a Range Bar Chart to show a comparison between several sets of data (for example, summaries of quantitative or time data). Each series is automatically colored differently for easier reading. The Range Bar Chart is similar to the [Bar Chart]({%slug components/chart/types/bar%}), which can be regarded as a Range Bar Chart with a "low" ("from") value of zero.
```

@[template](/_contentTemplates/chart/link-to-basics.md#understand-basics-and-databinding-first)

## Creating Blazor Range Bar Chart

1. Add a `ChartSeries` to the `ChartSeriesItems` collection.

> 2. Set its `Type` property to `ChartSeriesType.RangeBar`.

Reviewer: ambiguous "its"

> 2. Set its `Type` property to `ChartSeriesType.RangeBar`.

Author: ```suggestion
2. Set the series `Type` parameter to `ChartSeriesType.RangeBar`.
```

> 3. Provide a data collection to its `Data` property. You can use a [collection of arrays](#bind-range-column-series-to-collection-of-arrays) or a [collection of custom objects](#bind-range-column-series-to-custom-objects).

Reviewer: ambiguous "its"

> 3. Provide a data collection to its `Data` property. You can use a [collection of arrays](#bind-range-column-series-to-collection-of-arrays) or a [collection of custom objects](#bind-range-column-series-to-custom-objects).

Reviewer: ```suggestion
3. Provide a data collection to its `Data` property. You can use a [collection of arrays](#binding-range-column-series-to-collection-of-arrays) or a [collection of custom objects](#binding-range-column-series-to-custom-objects).
```

> 3. Provide a data collection to its `Data` property. You can use a [collection of arrays](#bind-range-column-series-to-collection-of-arrays) or a [collection of custom objects](#bind-range-column-series-to-custom-objects).

Author: ```suggestion
3. Provide a data collection to the series `Data` parameter. You can use a [collection of arrays](#bind-range-column-series-to-collection-of-arrays) or a [collection of custom objects](#bind-range-column-series-to-custom-objects).
```

4. If the Range Bar data is a collection of arrays, provide data for the `Categories` parameter of the `ChartCategoryAxis`.

> ### Bind Range Bar Series to Collection of Arrays

Reviewer: ```suggestion
### Binding Range Bar Series to Collection of Arrays
```

> In this case, set the `ChartSeries` `Data` parameter to a `List` of arrays or a jagged array (an array of arrays). The inner arrays should have two members - one for the "low" ("from") value, and one for the "high" ("to") value.

Reviewer: ```suggestion
1. Set the `ChartSeries` `Data` parameter to a `List` of arrays or a jagged array (an array of arrays). The inner arrays must have two members—one for the "low" ("from") value, and one for the "high" ("to") value.
```

> Set the `ChartSeries` `Data` parameter to a `List` of arrays or a jagged array (an array of arrays). The inner arrays must have two members—one for the "low" ("from") value, and one for the "high" ("to") value.

Author: ```suggestion
Set the `ChartSeries` `Data` parameter to a `List` of arrays or a jagged array (an array of arrays). The inner arrays must have two members—a lower one for the `from` value, and a higher one for `to` value.
```

> In addition, set the `Categories` parameter of the `ChartCategoryAxis` to `object[]`. The members of this array will be used as labels for the category axis in their respective order.

Reviewer: ```suggestion
1. Set the `Categories` parameter of the `ChartCategoryAxis` to `object[]`. The members of this array will be used as labels for the category axis in their respective order.
```

````CSHTML
<TelerikChart>
    <ChartTitle Text="Sleep Hours by Age"></ChartTitle>

    <ChartSeriesItems>
        <ChartSeries Type="ChartSeriesType.RangeBar" Data="@SleepData">
        </ChartSeries>
    </ChartSeriesItems>

    <ChartCategoryAxes>
        <ChartCategoryAxis Categories="@Categories"></ChartCategoryAxis>
    </ChartCategoryAxes>

    <ChartValueAxes>
        <ChartValueAxis Min="4"></ChartValueAxis>
    </ChartValueAxes>

    <ChartTooltip Visible="true">
        <Template>
            @{
                var dataItem = (int[])context.DataItem;
            }
            <span>@dataItem[0] - @dataItem[1] hours</span>
        </Template>
    </ChartTooltip>

</TelerikChart>

@code {
    private List<int[]> SleepData = new List<int[]>() {
        new int[] { 14, 17 },
        new int[] { 12, 16 },
        new int[] { 11, 14 },
        new int[] { 10, 13 },
        new int[] { 9, 12 },
        new int[] { 8, 10 },
        new int[] { 7, 9 },
        new int[] { 7, 8 }
    };

    private string[] Categories = new string[] {
        "0–3 m", "4-12 m", "1-2 y", "3-5 y", "6-12 y", "13-18 y", "18-60 y", "60+ y"
    };
}
````

### Bind Range Column Series to Custom Objects

> In this case, set the `ChartSeries` `Data` parameter to an `IEnumerable<T>`. Then, set the `FromField`, `ToField` and `CategoryField` parameters of the `ChartSeries` to properties of the `T` type.

Reviewer: ```suggestion
1. Set the `ChartSeries` `Data` parameter to an `IEnumerable<T>`. 
1. Set the `FromField`, `ToField`, and `CategoryField` parameters of the `ChartSeries` to properties of the `T` type.
```

            @{
                var dataItem = (RangeBarModel)context.DataItem;
            }
            <span>@dataItem.LowValue - @dataItem.HighValue hours</span>
        </Template>
    </ChartTooltip>

</TelerikChart>

@code {
    private List<RangeBarModel> SleepData = new List<RangeBarModel>() {
        new RangeBarModel { LowValue = 14, HighValue = 17, AgeGroup = "0-3 m" },
        new RangeBarModel { LowValue = 12, HighValue = 16, AgeGroup = "4-12 m" },
        new RangeBarModel { LowValue = 11, HighValue = 14, AgeGroup = "1-2 y" },
        new RangeBarModel { LowValue = 10, HighValue = 13, AgeGroup = "3-5 y" },
        new RangeBarModel { LowValue = 9, HighValue = 12, AgeGroup = "6-12 y" },
        new RangeBarModel { LowValue = 8, HighValue = 10, AgeGroup = "13-18 y" },
        new RangeBarModel { LowValue = 7, HighValue = 9, AgeGroup = "18-60 y" },
        new RangeBarModel { LowValue = 7, HighValue = 8, AgeGroup = "60+ y" }
    };

    public class RangeBarModel
    {
        public string AgeGroup { get; set; }
        public int LowValue { get; set; }
        public int HighValue { get; set; }
    }
}
````

## Range Bar Chart Specific Appearance Settings

> 

Reviewer: Use text between a heading and its subheading.

### Color

> The color of a series is controlled through the `Color` property that can take any valid CSS color (for example, `#abcdef`, `#f00`, or `blue`). The color control the fill color of the area.

Reviewer: The first word "color" in the second sentence refers to...?
```suggestion
The color of a series is controlled through the `Color` property that can take any valid CSS color (for example, `#abcdef`, `#f00`, or `blue`). The color controls the fill color of the area.
```
